types/src/traits.rs

```diff
@@ -81,3 +83,57 @@ pub trait SubscriptionClient: Client {
 	where
 		Notif: DeserializeOwned;
 }
+
+/// Marker trait for types that can be serialized as JSON array/sequence.
+///
+/// If your type isn't a sequence such as `String`, `usize` or similar.
+/// You could insert it in a tuple, slice, array or Vec for it to work.
+pub trait ToRpcParams: Serialize {
+	/// Serialized the type as a JSON array.
+	fn to_rpc_params(&self) -> Result<Box<RawValue>, serde_json::Error> {
+		serde_json::to_string(&self).map(|json| RawValue::from_string(json).expect("JSON String; qed"))
+	}
+}
+
+impl<P: Serialize> ToRpcParams for &[P] {}
+
+impl<P: Serialize> ToRpcParams for Vec<P> {}
+
+macro_rules! array_impls {
```

Comment:
const generics doesn't work because of `serde` I think.

Reply:
Ah, came here to ask about this.